tests/background/currentpoller_test.py

```diff
@@ -0,0 +1,261 @@
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from lsst.ts.rubintv.background.currentpoller import CurrentPoller
+from lsst.ts.rubintv.models.models import Camera, Location, get_current_day_obs
+from lsst.ts.rubintv.models.models_helpers import find_first
+from lsst.ts.rubintv.models.models_init import ModelsInitiator
+from moto import mock_s3
+
+from ..mockdata import RubinDataMocker
+
+m = ModelsInitiator()
+
+
+@pytest.fixture(scope="function")
+def rubin_data_mocker(mock_s3_client: Any) -> Iterator[RubinDataMocker]:
+    with mock_s3():
+        mocker = RubinDataMocker(m.locations, s3_required=True)
+        yield mocker
+
+
+@pytest.fixture
+def current_poller(rubin_data_mocker: RubinDataMocker) -> CurrentPoller:
+    return CurrentPoller(m.locations)
+
+
+@pytest.fixture(scope="function")
+def c_poller_no_mock_data(rubin_data_mocker: RubinDataMocker) -> Any:
+    with mock_s3():
+        yield CurrentPoller(m.locations)
+
+
+@pytest.mark.asyncio
+async def test_poll_buckets_for_todays_data(
+    current_poller: CurrentPoller, rubin_data_mocker: RubinDataMocker
+) -> None:
+    # Mocking external functions
+    with (
+        patch(
+            "lsst.ts.rubintv.models.models.get_current_day_obs",
+            return_value="2024-03-28",
+        ) as mock_day_obs,
+        patch(
+            "lsst.ts.rubintv.background.currentpoller.CurrentPoller.clear_all_data",
+            new_callable=AsyncMock,
+        ),
+        patch(
+            "lsst.ts.rubintv.background.currentpoller.CurrentPoller.sieve_out_metadata",
+            new_callable=AsyncMock,
+        ) as mock_metadata,
+        patch(
+            (
+                "lsst.ts.rubintv.background.currentpoller.CurrentPoller."
+                "sieve_out_night_reports"
+            ),
+            new_callable=AsyncMock,
+        ) as mock_night_reports,
+        patch(
+            "lsst.ts.rubintv.background.currentpoller.CurrentPoller."
+            "process_channel_objects",
+            new_callable=AsyncMock,
+        ) as mock_process_objects,
+    ):
+        # Execute test
+        try:
+            # Run the method for a specified number of seconds, then timeout
+            await asyncio.wait_for(
+                current_poller.poll_buckets_for_todays_data(), timeout=0.2
+            )
+        except asyncio.TimeoutError:
+            pass
+
+        assert mock_day_obs
+        mock_metadata.assert_called()
+        mock_night_reports.assert_called()
+        mock_process_objects.assert_called()
+        assert current_poller.completed_first_poll is True
+
+
+@pytest.mark.asyncio
+async def test_poll_buckets_for_today_process_and_store_seq_events(
+    current_poller: CurrentPoller, rubin_data_mocker: RubinDataMocker
+) -> None:
+    try:
+        await asyncio.wait_for(
+            current_poller.poll_buckets_for_todays_data(), timeout=0.3
+        )
+    except asyncio.TimeoutError:
+        # The timeout error is expected
+        pass
```

Comment:
This is likely not a good way for the testing to work- it's basically stopping the execution of the infinite loop background process with the timeout rather than there being a counter in the loop itself that can cause a break. But that's why it says it's expected, not that the exception needs testing.